File: greedy/prob1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        
        n = len(s)
        ans = 0    
        lc, rc = 0, 0
    
        start, end = 0, 1  # s[start:end]
        while end < n:
            if s[start] == "L":
                lc += 1
            else:
                rc += 1
            while lc != rc:
                if s[end] == "L":
                    lc += 1
                else:
                    rc += 1
                end += 1
            ans += 1
            start = end
            end = start + 1
            lc = 0
            rc = 0
            
        return ans

File: strings/prob1221.kt
// 1221 Recursion
//
class Solution {
    fun balancedStringSplit(s: String): Int {
        
        val n = s.length
        if (n == 0) {
            return 0
        }
        
        var cl = 0 
        var cr = 0
        if (s[0] == 'R') {
            cr++
        } else {
            cl++
        }
        var i = 0
        while (cr != cl && i < n-1) {
            i++
            if (s[i] == 'R') {
                cr++
            } else {
                cl++
            }
        }
        
        return 1 + balancedStringSplit(s.substring((i+1)..(n-1)))
        
    }
}
